gemrb/core/GUI/Window.cpp

```diff
@@ -18,452 +18,587 @@
  *
  */
 
-#include "GUI/Window.h"
+#include "Window.h"
 
-#include "GUI/Button.h"
-#include "GUI/MapControl.h"
-#include "GUI/ScrollBar.h"
+#include "GUI/GUIScriptInterface.h"
 
-#include "win32def.h"
-
-#include "ie_cursors.h"
+#include "Interface.h"
+#include "ScrollBar.h"
 
 namespace GemRB {
 
-Window::Window(unsigned short WindowID, unsigned short XPos,
-	unsigned short YPos, unsigned short Width, unsigned short Height)
-{
-	this->WindowID = WindowID;
-	this->XPos = XPos;
-	this->YPos = YPos;
-	this->Width = Width;
-	this->Height = Height;
-	this->BackGround = NULL;
-	lastC = NULL;
-	lastFocus = NULL;
-	lastMouseFocus = NULL;
-	lastOver = NULL;
-	Visible = WINDOW_INVISIBLE;
-	Flags = WF_CHANGED;
-	Cursor = IE_CURSOR_NORMAL;
-	DefaultControl[0] = -1;
-	DefaultControl[1] = -1;
-	ScrollControl = -1;
-	FunctionBar = false;
-	keyPressHandler = NULL;
-}
-
-Window::~Window()
-{
-	for (std::vector<Control*>::iterator m = Controls.begin(); m != Controls.end(); ++m) {
-		delete *m;
-	}
-	Controls.clear();
-	Sprite2D::FreeSprite( BackGround );
-	BackGround = NULL;
-}
-/** Add a Control in the Window */
-void Window::AddControl(Control* ctrl)
-{
-	if (ctrl == NULL) {
-		return;
-	}
-	ctrl->Owner = this;
-	for (std::vector<Control*>::iterator m = Controls.begin(); m != Controls.end(); ++m) {
-		if ((*m)->ControlID == ctrl->ControlID) {
-			ControlRemoved(*m);
-			delete *m;
-			*m = ctrl;
-			Invalidate();
-			return;
-		}
+Window::Window(const Region& frame, WindowManager& mgr)
+	: ScrollView(frame), manager(mgr)
+{
+	focusView = NULL;
+	trackingView = NULL;
+	hoverView = NULL;
+	backBuffer = NULL;
```

Comment:
Codacy found an issue: [Variable 'backBuffer' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/manual/gemrb/gemrb/pullRequest?prid=6310305)